cpp/include/cudf/strings/detail/copy_range.cuh

```diff
@@ -0,0 +1,217 @@
+                                   // TODO: first & last are used for the fill
+                                   // function for strings (instead of begin &
+                                   // end)
+                                   size_type target_begin, size_type target_end,
+                                   rmm::mr::device_memory_resource* mr =
+                                     rmm::mr::get_default_resource(),
+                                   cudaStream_t stream = 0) {
+  CUDF_EXPECTS((target_begin >= 0) &&
+                 (target_begin < target.size()) &&
+                 (target_end <= target.size()),
+               "Range is out of bounds.");
+
+  if (target_end != target_begin) {
+    auto p_target_device_view =
+      column_device_view::create(target.parent(), stream);
+    auto d_target = *p_target_device_view;
+
+    // create resulting null mask
+
+    std::pair<rmm::device_buffer, size_type> valid_mask{};
+    if (target.has_nulls()) {  // check validities for both source & target
+      valid_mask = cudf::experimental::detail::valid_if(
+        thrust::make_counting_iterator<size_type>(0),
+        thrust::make_counting_iterator<size_type>(target.size()),
+        [source_validity_begin, d_target, target_begin, target_end] __device__
+            (size_type idx) {
+          return (idx >= target_begin && idx < target_end) ?
+            *(source_validity_begin + (idx - target_begin)) :
+            d_target.is_valid(idx);
+        },
+        stream, mr);
+    }
+    else {  // check validities for source only
+      valid_mask = cudf::experimental::detail::valid_if(
+        thrust::make_counting_iterator<size_type>(0),
+        thrust::make_counting_iterator<size_type>(target.size()),
+        [source_validity_begin, d_target, target_begin, target_end] __device__
+            (size_type idx) {
+          return (idx >= target_begin && idx < target_end) ?
+            *(source_validity_begin + (idx - target_begin)) : true;
+        },
+        stream, mr);
+    }
+
+    auto null_count = valid_mask.second;
+    rmm::device_buffer null_mask{};
+    if (target.parent().nullable() ||
+        null_count > 0) {
+      null_mask = std::move(valid_mask.first);
+    }
+
+    // build offsets column
+
+    std::unique_ptr<column> p_offsets_column{nullptr};
+    if (target.has_nulls()) {  // check validities for both source & target
+      auto string_size_begin =
+        thrust::make_transform_iterator(
+          thrust::make_counting_iterator(0),
+          [source_value_begin, source_validity_begin, d_target, target_begin,
+              target_end] __device__ (size_type idx) {
+            if (idx >= target_begin && idx < target_end) {
+              return *(source_validity_begin + (idx - target_begin)) ?
+                (*(source_value_begin + (idx - target_begin))).size_bytes() : 0;
+            }
+            else {
+              return d_target.is_valid(idx) ?
```

Comment:
You already know the target contains nulls, so you can use the faster `nocheck` version. 
```suggestion
              return d_target.is_valid_nocheck(idx) ?
```